File: samples/course23/ch2/rednote-ui/src/dto/admin-dashboard-dto.ts
import type { NoteBrowseCountDto } from "./note-browse-count-dto";
import type { NoteBrowseTimeDto } from "./note-browse-time-dto";

export interface AdminDashboardDto { 
  userCount: number;
  noteCount: number;
  commentCount: number;
  noteBrowseCountDtoList: Array<NoteBrowseCountDto>;
  noteBrowseTimeDtoList: Array<NoteBrowseTimeDto>;
}

File: samples/course23/ch3/rednote-microservices/rednote-common/src/main/java/com/waylau/rednote/common/dto/AdminDashboardDto.java
package com.waylau.rednote.common.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

/**
 * AdminDashboardDto
 *
 * @author <a href="https://waylau.com">Way Lau</a>
 * @version 2025/12/18
 **/
@Getter
@Setter
@AllArgsConstructor
public class AdminDashboardDto {
    private long userCount;
    private long noteCount;
    private long commentCount;
    private List<NoteBrowseCountDto> noteBrowseCountDtoList;
    private List<NoteBrowseTimeDto> noteBrowseTimeDtoList;
}
